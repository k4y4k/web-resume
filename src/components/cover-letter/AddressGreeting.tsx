import "twin.macro";
import * as React from "react";
import { graphql, useStaticQuery } from "gatsby";

interface AddressGreetingProps {
  letterContents: {
    childMarkdownRemark: {
      frontmatter: {
        recruiterFirstName: string;
        recruiterLastName: string;
        address: string;
        city: string;
        countryCode: string;
        postalCode: string;
        region: string;
      };
    };
  };
}

const AddressGreeting = (): JSX.Element => {
  const { letterContents }: AddressGreetingProps = useStaticQuery(graphql`
    {
      letterContents: file(extension: { eq: "md" }, name: { eq: "letter" }) {
        childMarkdownRemark {
          frontmatter {
            recruiterFirstName
            recruiterLastName
            address
            city
            countryCode
            postalCode
            region
          }
        }
      }
    }
  `);

  const {
    recruiterFirstName,
    recruiterLastName,
    address,
    city,
    region,
    postalCode,
    countryCode,
  } = letterContents.childMarkdownRemark.frontmatter;

  return (
    <>
      <p tw="pb-0!">
        {recruiterFirstName}{" "}
        {recruiterLastName === "_" ? null : recruiterLastName}
      </p>
      <p tw="pb-0!">{address}</p>
      <p>
        {city !== " " ? `${city}, ` : null}
        {region !== " " ? `${region}, ` : null}
        {postalCode !== " " ? `${postalCode}, ` : null}
        {countryCode !== " " ? `${countryCode}` : null}
      </p>

      <h1 tw="mt-4 mb-1 font-bold text-xl">Dear {recruiterFirstName},</h1>
    </>
  );
};

export default AddressGreeting;
