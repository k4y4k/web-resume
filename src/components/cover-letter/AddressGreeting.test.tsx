import * as React from 'react'
import AddressGreeting from './AddressGreeting'
import { render } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'

describe('<AddressGreeting/>', () => {
  beforeAll(() =>
    (useStaticQuery as jest.Mock).mockReturnValue({
      jsonName: {
        childDataJson: {
          basics: {
            name: 'Blaze Powlowski',
          },
        },
      },
      letterContents: {
        childMarkdownRemark: {
          frontmatter: {
            recruiterName: 'Polycarp Medvedev',
            address: '60342 Athena Plaza',
            city: 'Port Idellabury',
            countryCode: 'ES',
            postalCode: '60033-4381',
            region: 'South Dakota',
          },
          html: '<p>esse dolore molestias suscipit dignissimos illum cupiditate consequatur non voluptates ut dolores aliquam cum qui sed libero architecto iusto porro unde suscipit non non tempore officiis vero aut rem id repellendus et iste</p>\n<p>voluptatem beatae illo rerum mollitia expedita et ducimus aliquam deserunt aut suscipit error facere recusandae sunt in aut aperiam ut quia minima ullam quasi laudantium alias velit accusamus</p>\n<p>voluptatem maiores repellendus voluptate aperiam quia magni quisquam incidunt quis voluptas repudiandae aut dolorum eum</p>\n<p>voluptas consequatur ab reiciendis omnis ipsam qui quia aperiam dolores ut ducimus rerum magnam voluptatem dolorum natus aut minus odio reprehenderit impedit quis recusandae repellat et autem nihil voluptas beatae minima laboriosam cum in sed iusto officia</p>\n<p>sunt laborum hic rerum quaerat vel quidem perspiciatis et et porro iste laborum reiciendis enim nostrum ut</p>\n<p>tenetur omnis ad at ratione quia quia et alias sequi sunt aliquid eos aut aut odio eligendi consequatur nulla optio tenetur tempore quaerat et pariatur harum quasi ipsam optio molestiae sed consectetur quos tenetur nihil velit</p>\n<p>quis similique libero nemo optio et voluptatum similique ut</p>\n<p>voluptas nulla aut similique possimus nihil sed laboriosam dolorum molestiae quia voluptatem consequuntur corporis impedit dolorum quos incidunt ut numquam sit aut error vero consequatur</p>',
        },
      },
    })
  )

  test('renders OK', () => {
    const { container } = render(<AddressGreeting />)

    expect(container).toMatchSnapshot()
  })
})
