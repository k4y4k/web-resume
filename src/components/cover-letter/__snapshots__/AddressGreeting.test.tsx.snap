// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressGreeting/> renders OK 1`] = `
.c0 {
  padding-bottom: 0px !important;
}

.c1 {
  padding-bottom: 0px !important;
}

<div>
  <p
    class="c0"
  >
    Polycarp Medvedev
  </p>
  <p
    class="c1"
  >
    60342 Athena Plaza
  </p>
  <p>
    Port Idellabury
    , 
    South Dakota
    , 
    60033-4381
    , 
    ES
  </p>
  <p>
    Dear 
    Polycarp Medvedev
    ,
  </p>
</div>
`;
